Skeleton/src/Volunteer.cpp

```diff
@@ -1 +1,263 @@
+{
+    return distancePerStep;
+} 
+bool DriverVolunteer::decreaseDistanceLeft() //Decrease distanceLeft by distancePerStep,return true if distanceLeft<=0,false otherwise
+{
+    if(distanceLeft > 0){
+        distanceLeft = distanceLeft - distancePerStep;
+    }
+    return (distanceLeft <= 0);
+} 
+bool DriverVolunteer::hasOrdersLeft() const 
+{
+    return true;
+}
+bool DriverVolunteer::canTakeOrder(const Order &order) const 
+{
+    return (order.getStatus() == OrderStatus::COLLECTING && !Volunteer::isBusy() && order.getDistance() <= maxDistance); // Signal if the volunteer is not busy and the order is within the maxDistance
+} 
+
+void DriverVolunteer::acceptOrder(const Order &order)  
+{
+    if(canTakeOrder(order)){
+        distanceLeft = order.getDistance(); // Assign distanceLeft to order's distance
+        activeOrderId = order.getId();
+    }
+} 
+
+void DriverVolunteer::step() 
+{ 
+    if(decreaseDistanceLeft()){
+        completedOrderId = activeOrderId;
+        activeOrderId = NO_ORDER;
+    }
+} 
+
+string DriverVolunteer::toString() const {
+        return "Driver Volunteer - ID: " + std::to_string(getId())
+     + ", Name: " + getName() 
+     + ", Active Order ID: " + std::to_string(getActiveOrderId()) 
+     + ", Completed Order ID: " + std::to_string(getCompletedOrderId())
+     + ", Maximum Distance: " + std::to_string(getMaxDistance())
+     + ", Distance Left: " + std::to_string(getDistanceLeft())
+     + ", Distance per Step: " + std::to_string(getDistancePerStep());
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////
+/////////////////////////////// Limited Driver Collector ///////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////
+
+LimitedDriverVolunteer::LimitedDriverVolunteer(int id, const string &name, int maxDistance, int distancePerStep,int maxOrders) : DriverVolunteer(id, name, maxDistance, distancePerStep) , maxOrders(maxOrders), ordersLeft(maxOrders) {}
+LimitedDriverVolunteer* LimitedDriverVolunteer::clone() const
+{
+    return new LimitedDriverVolunteer(*this);
+}
+
+int LimitedDriverVolunteer::getMaxOrders() const 
+{
+    return maxOrders;
+}
+
+int LimitedDriverVolunteer::getNumOrdersLeft() const 
+{
+    return ordersLeft;
+}
+
+bool LimitedDriverVolunteer::hasOrdersLeft() const
+{
+    return (ordersLeft > 0);
+}
+
+bool LimitedDriverVolunteer::canTakeOrder(const Order &order) const {
+    return (hasOrdersLeft() && order.getStatus() == OrderStatus::COLLECTING && !Volunteer::isBusy() && order.getDistance() <= DriverVolunteer::getMaxDistance());
+} // Signal if the volunteer is not busy, the order is within the maxDistance.
+
+void LimitedDriverVolunteer::acceptOrder(const Order &order) {
+    if(canTakeOrder(order))
+    {
+        ordersLeft--;
+        activeOrderId = order.getId();
+        setDistanceLeft(order.getDistance());
+    }
```

Comment:
missing step function i think

Reply:
I think that the step() function in LimitedDriverVolunteer in not necessary, because it will implement it via DriverVolunteer

Reply:
ok you are right, but it is a bit dangerous, for example-
i want in every step to check canTakeOrder, so in that case i think it will not implement the limited check, only the regular